File: service/src/main/java/com/boxfox/cross/service/friend/FriendUsecase.java
package com.boxfox.cross.service.friend;

import com.linkbit.android.entity.UserModel;
import io.reactivex.Completable;
import io.reactivex.Single;
import java.util.List;

public interface FriendUsecase {

  Single<List<UserModel>> loadFriends(String uid);

  Single<UserModel> getUser(String uid);

  Completable addFriend(String ownUid, String uid);

  Completable deleteFriend(String ownUid, String uid);

  Single<List<UserModel>> serachUsers(String text);
}

File: service/src/main/java/com/boxfox/cross/service/friend/FriendUsecase.kt
package com.boxfox.cross.service.friend

import com.boxfox.cross.common.RoutingException
import com.boxfox.cross.common.entity.UserModel
import org.jooq.DSLContext

interface FriendUsecase {

    @Throws(RoutingException::class)
    fun loadFriends(ctx: DSLContext, uid: String): List<UserModel>

    @Throws(RoutingException::class)
    fun getUser(ctx: DSLContext, uid: String): UserModel

    @Throws(RoutingException::class)
    fun addFriend(ctx: DSLContext, ownUid: String, uid: String)

    @Throws(RoutingException::class)
    fun deleteFriend(ctx: DSLContext, ownUid: String, uid: String)

    @Throws(RoutingException::class)
    fun searchUser(ctx: DSLContext, text: String): List<UserModel>
}
